plugins/secretstores/gdchauth/gdchauth.go

```diff
@@ -0,0 +1,214 @@
+package gdchauth
+
+import (
+	"bytes"
+	"context"
+	"crypto/x509"
+	_ "embed"
+	"encoding/json"
+	"encoding/pem"
+	"errors"
+	"fmt"
+	"io"
+	"net/http"
+	"os"
+	"sync"
+	"time"
+
+	"github.com/golang-jwt/jwt/v5"
+
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/config"
+	common_http "github.com/influxdata/telegraf/plugins/common/http"
+	"github.com/influxdata/telegraf/plugins/secretstores"
+)
+
+//go:embed sample.conf
+var sampleConfig string
+
+func (*GdchAuth) SampleConfig() string {
+	return sampleConfig
+}
+
+// GdchAuth is the main authenticator struct
+type GdchAuth struct {
+	Audience           string          `toml:"audience"`
+	Log                telegraf.Logger `toml:"-"`
+	ServiceAccountFile string          `toml:"service_account_file"`
+	TokenExpiryBuffer  config.Duration `toml:"token_expiry_buffer"`
```

Comment:
Please do order those in a sensible way, most important ones go to the top...


```suggestion
	ServiceAccountFile string          `toml:"service_account_file"`
	Audience           string          `toml:"audience"`
	TokenExpiryBuffer  config.Duration `toml:"token_expiry_buffer"`
	Log                telegraf.Logger `toml:"-"`
```